app/scripts/eth-trustvault-keyring.js

```diff
@@ -0,0 +1,445 @@
+          enc: "${auth.enc}",
+          iv: "${auth.iv}",
+          tag: "${auth.tag}"
+        }
+      ) {
+        enc,
+        iv,
+        tag
+      }
+    }`
+  }
+
+  _getAuthTokenQuery (email, firstPinDigit, secondPinDigit, sessionToken) {
+    return `query {
+      getAuthenticationTokens(
+        getAuthenticationTokensInput: {
+          email: "${email}",
+          pinChallenge: {
+            firstPinDigit: "${firstPinDigit}",
+            secondPinDigit: "${secondPinDigit}",
+            sessionToken: "${sessionToken}"
+          }
+        }
+      ) {
+        authentication {
+          enc,
+          iv,
+          tag
+        },
+        pinChallenge {
+          firstPinDigitPosition,
+          secondPinDigitPosition,
+          sessionToken
+        }
+      }
+    }`
+  }
+
+  _getPartialPinChallengeQuery (email) {
+    return `query {
+      getPartialPinChallenge(email: "${email}") {
+        firstPinDigitPosition,
+        secondPinDigitPosition,
+        sessionToken
+      }
+    }`
+  }
+
+  _getSignTransactionMutation (auth, { tx, address, transactionDigest }) {
+    return `mutation {
+      requestSignature(
+        requestSignatureInput: {
+          authentication: {
+            enc: "${auth.enc}",
+            iv: "${auth.iv}",
+            tag: "${auth.tag}"
+          },
+          transaction: {
+            nonce: ${tx.nonce},
+            chainId: ${tx.chainId},
+            gasPrice: "${tx.gasPrice}",
+            gasLimit: "${tx.gasLimit}",
+            value: "${tx.value}",
+            fromAddress: "${address}",
+            ${
+  tx.data ? `data: "${tx.data}",` : ''
+}
+            to: "${tx.to}",
+            v:${tx.v}
+          },
+          transactionDigest: "${transactionDigest}",
+          source: "MetaMask"
+        }
+      ) {
+        transactionId
+      }
+    }`
+  }
+
+  _getTransactionInfoQuery (auth, requestContext) {
+    return `query {
+        userWallet(
+          authentication: {
+            enc: "${auth.enc}",
+            iv: "${auth.iv}",
+            tag:" ${auth.tag}"
+          }
+        ) {
+          getTransactionInfo(transactionId: "${requestContext.transactionId}") {
+            status
+            signedTransaction {
+              transactionDigest
+              transaction {
+                  to
+                  value
+                  chainId
+                  nonce
+                  gasLimit
+                  gasPrice
+                  fromAddress
+                  r
+                  s
+                  v
+                }
+              }
+              status
+            }
+          }
+        }`
+  }
+
+  /* Helper functions */
+
+  _normalize (buf) {
+    return ethUtil.bufferToHex(buf).toString()
+  }
+
+  _normalizeNum (buf) {
+    return new BigNumber(ethUtil.bufferToHex(buf)).toString(10)
+  }
+
+  _timeout (ms) {
+    return new Promise(resolve => setTimeout(resolve, ms))
+  }
+}
+
+TrustvaultKeyring.type = type
```

Comment:
Can we move this onto the class as a `static` field? We can also avoid the constant if we reference it through `TrustvaultKeyring.type`.